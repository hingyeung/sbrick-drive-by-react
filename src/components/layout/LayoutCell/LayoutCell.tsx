import * as React from 'react';
import * as classNames from 'classnames';

export interface Props {
  extraClassNames: string;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  children: any;
}

function isValidBootstrapColumnValue(value: number) {
  return value >= 1 && value <= 12;
}

function buildBootstrapGridColumnClasses(props: Props): any {
  return {
    [`col-sm-${props.sm}`]: props.sm && isValidBootstrapColumnValue(props.sm),
    [`col-md-${props.md}`]: props.md && isValidBootstrapColumnValue(props.md),
    [`col-lg-${props.lg}`]: props.lg && isValidBootstrapColumnValue(props.lg),
    [`col-xl-${props.xl}`]: props.xl && isValidBootstrapColumnValue(props.xl)
  };
}

const LayoutCell: React.SFC<Props> = (props: Props) => {
  return (
    <div className={classNames(props.extraClassNames, buildBootstrapGridColumnClasses(props))}>
      {props.children}
    </div>
  );
};

export default LayoutCell;