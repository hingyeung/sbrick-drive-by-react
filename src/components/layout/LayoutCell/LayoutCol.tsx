import * as React from 'react';
import * as classNames from 'classnames';
import { StatelessComponent } from 'react';

interface LayoutCellComponentProps {
  className?: string;
  componentClass?: any;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
}

const isValidBootstrapColumnValue = function(value: number) {
  return value >= 1 && value <= 12;
};

const buildBootstrapGridColumnClasses = function(props: any): any {
  return {
    [`col-sm-${props.sm}`]: props.sm && isValidBootstrapColumnValue(props.sm),
    [`col-md-${props.md}`]: props.md && isValidBootstrapColumnValue(props.md),
    [`col-lg-${props.lg}`]: props.lg && isValidBootstrapColumnValue(props.lg),
    [`col-xl-${props.xl}`]: props.xl && isValidBootstrapColumnValue(props.xl)
  };
};

const LayoutCol: StatelessComponent<LayoutCellComponentProps> = (props) => {
  const {
    componentClass: Component,
    className,
    ...otherProps
  } = props;

  return (
    <Component {...otherProps} className={classNames(className, 'col', buildBootstrapGridColumnClasses(otherProps))}/>
  );
};

LayoutCol.defaultProps = {
  componentClass: 'div'
};

export default LayoutCol;