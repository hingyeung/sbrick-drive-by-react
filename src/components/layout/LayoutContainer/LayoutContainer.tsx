import * as React from 'react';
import * as classNames from 'classnames';

export interface Props {
  extraClassNames: string;
  children: JSX.Element | JSX.Element[];
}

export default (props: Props) => {
  return (
    <div className={classNames(props.extraClassNames, 'container')}>
        {props.children}
    </div>
  );
};