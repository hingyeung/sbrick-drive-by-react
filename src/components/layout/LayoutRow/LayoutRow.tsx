import * as React from 'react';
import * as classNames from 'classnames';

interface Props {
  className?: string;
  componentClass?: any;
  propsForComponent?: object;
}

const LayoutRow: React.SFC<Props> = (props) => {
  const {
    componentClass: Component,
    propsForComponent,
    className,
    ...otherProps
  } = props;

  return (
    <Component {...otherProps} className={classNames(className, 'row')} {...propsForComponent}/>
  );
};

LayoutRow.defaultProps = {
  componentClass: 'div'
};

export default LayoutRow;