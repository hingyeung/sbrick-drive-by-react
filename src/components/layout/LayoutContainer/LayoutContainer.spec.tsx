import { shallow } from 'enzyme';
import * as React from 'react';
import LayoutContainer from './LayoutContainer';
import LayoutRow from '../LayoutRow/LayoutRow';
import LayoutCell from '../LayoutCell/LayoutCell';

describe('LayoutContainer', () => {
  it('should render extraClassNames in props', () => {
    const wrapper = shallow(
      <LayoutContainer extraClassNames="my-container">
        <LayoutRow extraClassNames="my-row">
          <LayoutCell extraClassNames="my-cell">hello</LayoutCell>
        </LayoutRow>
      </LayoutContainer>);

    expect(wrapper.find('.my-container')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});