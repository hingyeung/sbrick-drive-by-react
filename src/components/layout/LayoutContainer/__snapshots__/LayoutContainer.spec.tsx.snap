// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutContainer should render extraClassNames in props 1`] = `
<div
  className="my-container container"
>
  <Component
    extraClassNames="my-row"
  >
    <LayoutCell
      extraClassNames="my-cell"
    >
      hello
    </LayoutCell>
  </Component>
</div>
`;
